sdk/main/src/ED25519PrivateKey.cc

```diff
@@ -96,16 +104,25 @@ std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::generatePrivateKey()
 //-----
 std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::fromString(std::string_view key)
 {
-  if (key.size() != KEY_SIZE * 2 + DER_ENCODED_PREFIX_HEX.size() && key.size() != KEY_SIZE * 2)
+  std::string formattedKey = key.data();
```

Comment:
Same here, I don't think this function needs another copy of the `key` string, you can just use `key` itself.

Reply:
Key here is a string_view should be immutable. The class is designed to take in string_views. Thats why i'd rather have a copy than change the overall design. Anyway the copy will be made on a very tiny input in the main scenario.

Reply:
Ah right you're converting to hex at the end of the function anyways and rewriting. Yeah this is good